daemon/src/setup_taker.rs

```diff
@@ -189,6 +230,21 @@ impl xtra::Actor for Actor {
             tracing::warn!(id = %self.order_id, "Stopping setup_taker actor: {e}");
             ctx.stop()
         }
+
+        let maker_response_timeout = {
+            let this = ctx.address().expect("self to be alive");
+            async move {
+                tokio::time::sleep(MAKER_RESPONSE_TIMEOUT).await;
+
+                this.send(MakerResponseTimeoutReached {
+                    timeout: MAKER_RESPONSE_TIMEOUT,
+                })
+                .await
+                .expect("can send to ourselves");
+            }
+        };
```

Comment:
Note: An extension trait on `Address` for a delayed send would allow us to remove this duplication between the actors.

Reply:
I think we should go further and harmonize those actors. I don't think they should have the same overall corset for taker and maker, i.e. all taker protocols should be aligned and all maker protocols should be aligned.
All these protocols should follow the same pattern of taker proposes, maker accepts/rejects, message exchange, done.

Reply:
I am not sold on this.

1. As we are discussing in https://github.com/itchysats/itchysats/pull/1169, rollover will likely no longer have accept/reject.
2. Collaborative close might be replaced with "placing an order" and is thus the same as opening a position.
3. With combined positions, placing an order will need to be decoupled from protocol that runs on-chain. I think we will need some logic that figures out stuff like "If I am 1000 contracts long and place a sell order for 500 contract, I am only going to be 500 contracts long." I think this kind of book-keeping will need to be decoupled from what we are actually doing on-chain.

Long story short: These protocols only have the same structure because we invented them like that. I am no longer convinced that they should have this structure so I am hesitant to invest into abstractions that would make them be the same.

On the other hand, a generic extension trait like `Address::send_delayed` can be useful in many cases, will reduce the code duplication here and is unopinionated about our domain.

Reply:
https://github.com/itchysats/itchysats/issues/1316